### PR TITLE
wgsl: Add AbstractFloat `modf` execution tests

### LINKED ISSUE #2569: wgsl: AbstractFloat execution tests for `modf`
Add AbstractFloat execution tests for the `modf` builtin.
```
@const @must_use fn modf(e: T) -> __modf_result_abstract

T is AbstractFloat
```
Splits e into fractional and whole number parts.
The whole part is [trunc](https://www.w3.org/TR/WGSL/#trunc-builtin)(e), and the fractional part is e - [trunc](https://www.w3.org/TR/WGSL/#trunc-builtin)(e).

Returns the __modf_result_abstract built-in structure, defined as follows:

```
struct __modf_result_abstract {
  fract : AbstractFloat, // fractional part
  whole : AbstractFloat  // whole part
}
```

Accuracy is `Correctly rounded`

There is also a vector form of modf

See https://www.w3.org/TR/WGSL/#modf-builtin

### PR DESCRIPTION
**Pull request by zoddicus**
Fixes #2569

<hr>

**Requirements for PR author:**

- [x] All missing test coverage is tracked with "TODO" or `.unimplemented()`.
- [x] New helpers are `/** documented */` and new helper files are found in `helper_index.txt`.
- [x] Test behaves as expected in a WebGPU implementation. (If not passing, explain above.)

**Requirements for [reviewer sign-off](https://github.com/gpuweb/cts/blob/main/docs/reviews.md):**

- [ ] Tests are properly located in the test tree.
- [ ] [Test descriptions](https://github.com/gpuweb/cts/blob/main/docs/intro/plans.md) allow a reader to "read only the test plans and evaluate coverage completeness", and accurately reflect the test code.
- [ ] Tests provide complete coverage (including validation control cases). **Missing coverage MUST be covered by TODOs.**
- [ ] Helpers and types promote readability and maintainability.

When landing this PR, be sure to make any necessary issue status updates.

### CODE REVIEW

**Comment by zoddicus** [comment]
Stacked PR, new code is in the last diff, 8c59b531827e79f64aec5ebd7ebccfb781829e88

File: src/webgpu/shader/execution/expression/call/builtin/modf.spec.ts [hunk 1]
```diff
@@ -114,7 +132,7 @@ function vectorRange(kind: FPKind, dim: number): number[][] {
 }
 
 // Cases: [f32|f16]_[fract|whole]
```
Comment (jzm-intel) [inline src/webgpu/shader/execution/expression/call/builtin/modf.spec.ts:134]:
``` suggestion
// Cases: [f32|f16|abstract]_[fract|whole]
```

File: src/webgpu/shader/execution/expression/call/builtin/modf.spec.ts [hunk 2]
```diff
@@ -126,7 +144,7 @@ const scalar_cases = (['f32', 'f16'] as const)
   .reduce((a, b) => ({ ...a, ...b }), {});
 
 // Cases: [f32|f16]_vecN_[fract|whole]
```
Comment (jzm-intel) [inline src/webgpu/shader/execution/expression/call/builtin/modf.spec.ts:146]:
```suggestion
// Cases: [f32|f16|abstract]_vecN_[fract|whole]
```

**Review by jzm-intel — APPROVED** [review]
Looks good to me, thanks

**Pull request merged**